core/src/lib.rs

```diff
@@ -39,17 +39,32 @@
 
 #![warn(missing_docs)]
 #![warn(rust_2018_idioms)]
+#![cfg_attr(
+    all(feature = "nightly", target_vendor = "fortanix", target_env = "sgx"),
+    feature(sgx_platform)
```

Comment:
It's currently not really possible to consume the SGX target on stable. Other crates, e.g. [backtrace](https://github.com/alexcrichton/backtrace-rs/blob/master/src/lib.rs#L74) therefore just assume that you're on nightly when the target is SGX. I think I'd prefer that over the possibility here of accidentally using spinlocks.

Reply:
When used in `std` I modify all `cfg(feature = "nightly")` that I want to make use of into `cfg(any(feature = "nightly", feature = "i-am-libstd"))`. However, I don't want all features that the nightly `feature` activates when I use this lib in `std`. So I only modify a subset of them like that.

If it's reasonable to assume the only ones who use Rust on SGX use nightly and everything else should fail, then I can indeed remove this feature check and just assume nightly.

Reply:
In other words: `std` would never fall back to spinlocks on any Rust release channel. Because the integration into `std` would always activate the desired "nightly" features.